src/middlewared/middlewared/plugins/config.py

```diff
@@ -241,11 +242,29 @@ def validate_alembic(self, alembic_version):
     @job(lock='config_reset', logs=True)
     def reset(self, job, options):
         """
-        Reset database to configuration defaults.
+        Reset database to configuration defaults. Access to this endpoint is restricted to
+        either root or admin accounts that were set up during initial install.
 
         If `reboot` is true this job will reboot the system after its completed with a delay of 10
         seconds.
+
+        NOTE: In case of clustered TrueNAS this may unrecoverably destroy cluster-wide
+        configuration.
         """
+        job.set_progress(0, 'Performing credential check')
+        if job.credentials is not None and job.credentials.is_user_session:
```

Comment:
What does it mean if `job.credentials is None` ?

Reply:
Internal job. I suppose here it would be rather unexpected since we shouldn't have anyone writing a method that calls into this.